main.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"database/sql"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/gorilla/mux"
+)
+
+//===========================================================================================
+//===========================================================================================
+//MODEL
+var db *sql.DB
+var err error
+
+/*create table users(Username varchar(10) primary,
+FirstName varchar(10),LastName varchar(10),Password varchar(20),
+Gender varchar(10),Country varchar(20),Age varchar(10)); */
+
+//User struct
+type User struct {
+	Username  string `json:"username"`
+	FirstName string `json:"firstname"`
+	LastName  string `json:"lastname"`
+	Password  string `json:"password"`
+	Gender    string `json:"gender"`
+	Country   string `json:"country"`
+	Age       string `json:"age"`
+	Email     string `json:"email"`
+}
+
+//===========================================================================================
+//CONTROLLERSb
+func handleSignUp(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	var user User
+	stmt, err := db.Prepare("INSERT INTO users(Username,FirstName,LastName,Password,Gender,Country,Age,Email) VALUES(?)")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// keyVal := make(map[string]string)
+	err = json.Unmarshal(body, &user)
+	if err != nil {
+		log.Fatal(err)
+	}
+	// Username := keyVal["username"]
+	// FirstName := keyVal["firstname"]
+	// LastName := keyVal["lastname"]
+	// Password := keyVal["password"]
+	// Gender := keyVal["gender"]
+	// Country := keyVal["country"]
+	// Age := keyVal["age"]
+	// Email := keyVal["email"]
+
+	_, err = stmt.Exec(user.Username, user.FirstName, user.LastName, user.Password, user.Gender, user.Country, user.Age, user.Email)
+	if err != nil {
+		log.Fatal(err)
+	}
+	log.Print("new user is created.")
+}
+
+// func handleSignIn(w http.ResponseWriter, r *http.Request) {

```

Comment:
Don't comment old code. Just commit them, and then remove completely. This keeps the code clean, easy to read and you can checkout old commit to see your old code, if needed.